include/onnxruntime/core/providers/dml/dml_provider_factory.h

```diff
@@ -30,6 +30,23 @@
 extern "C" {
 #endif
 
+enum class OrtDmlPerformancePreference {
+  Default = 0,
+  HighPerformance = 1,
+  LowPower = 2
+};
+
+enum class OrtDmlDeviceFilter {
```

Comment:
>class OrtDmlDeviceFilter [](http://example.com/codeflow?start=5&length=24)

probably should just be :

enum OrtDmlDeviceFilter {
  Gpu = 1,
  Npu = 2,
}

Reply:
not enum class*

Reply:
Then you can do this:
auto filter = OrtDmlDeviceFilter::Gpu | OrtDmlDeviceFilter::Npu;